pkg/server/apiutil/rangeutil.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package apiutil
+
+import (
+	"context"
+	"math"
+
+	"github.com/cockroachdb/cockroach/pkg/keys"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descs"
+	"github.com/cockroachdb/errors"
+)
+
+type IndexNames struct {
+	Database string
+	Table    string
+	Index    string
+	Span     roachpb.Span
+}
+
+// Equal only compares the names, not the spans
+func (idx IndexNames) Equal(other IndexNames) bool {
+	return idx.Database == other.Database &&
+		idx.Table == other.Table &&
+		idx.Index == other.Index
+}
+
+// GetRangeIndexMappings translates a set of ordered ranges into a
+// RangeID -> []IndexNames mapping. It does this by executing the fololowing steps:
+//  1. Convert the set of ranges to a set of spans.
+//  2. Get the table descriptors that fall within the given spans.
+//  3. Get the database, table and index name for all indexes found in the descriptors.
+//  4. Return a mapping of the indexes which appear in each range.
+func GetRangeIndexMapping(
```

Comment:
Primary Entrypoint for callers: A function in which a set of ranges can be
passed in and a mapping from those ranges to indexes can be
returned.